app/src/main/java/com/example/plantsapp/presentation/ui/plantcreation/PlantCreationViewModel.kt

```diff
@@ -31,12 +35,16 @@ class PlantCreationViewModel(
                 Plant(
                     Plant.Name(plantName),
                     speciesName,
-                    "",
+                    selectedPicture.value ?: Uri.EMPTY,
                     wateringFrequency.toInt()
                 )
             )
 
             _toNavigateBack.value = Event(Unit)
         }
     }
+
+    fun onImageSelected(uri: Uri?) {
```

Comment:
here you can change to onImageSelected(uri: Uri) 
because your viewmodel doesn't want to know when somebody is trying to give it null - it just want to know what value you've passed here (bot some value != null)